Source/WarpX.cpp

```diff
@@ -1487,13 +1491,40 @@ WarpX::AllocLevelMFs (int lev, const BoxArray& ba, const DistributionMapping& dm
     if (deposit_charge)
     {
         rho_fp[lev] = std::make_unique<MultiFab>(amrex::convert(ba,rho_nodal_flag),dm,2*ncomps,ngRho,tag("rho_fp"));
+
+        // Also allocate the MultiFab for the full phi grid
+        // get the full domain
+        Box domain_box = Geom(lev).Domain();
+        // make a new BoxArray from the domain Box
+        BoxArray ba_full(domain_box);
+        // make a DistributionMapping from the new BoxArray
+        DistributionMapping dm_full;
+        // force that distribution mapping to only go to the root proc
+        Vector<int> pmap = {0};
+        dm_full.define(pmap);
+        // make a FabArray to hold the phi data
+        full_rho_fp[lev] = std::make_unique<MultiFab>(amrex::convert(ba_full,rho_nodal_flag),dm_full,2*ncomps,ngRho,tag("full_rho_fp"));
```

Comment:
Down the line, if we want to get some or all of this code upstream, unconditional allocation of full-domain-sized arrays will probably have to be changed.